lib/device_detector/metadata_extractor.rb

```diff
@@ -0,0 +1,37 @@
+class DeviceDetector
+  class MetadataExtractor < Struct.new(:user_agent, :regex_meta)
+
+    def call
+      regex_meta.any? ? extract_version : nil
+    end
+
+    private
+
+    def metadata_string
+      fail "#{self.name} (a child of MetadataExtractor) must implement the '#{__method__}' method."
+    end
+
+    def extract_version
```

Comment:
I guess we need to rename this method and `replace_version_string_with`. Ideas for a better name?